### ISSUE TITLE
urlpreview: Don't fetch previews for URLs in code blocks

### ISSUE THREAD

**Issue by peterrus** [coffeebank/coffee-maubot#5]
When someone embeds an URL in a code block (backtick or triple backtick) we probably shouldn't try to fetch a preview because that URL is there for different purposes than a regular URL that is posted to a channel.

**Comment by coffeebank** [comment]
Thinking a possible way is to run two Regex commands:

1. The first one remains the same, parsing all instances of https:// 
2. The second one parses for all instances of (\`|\`\`\`) ... https:// ... (\`|\`\`\`) and returning the urls

Then, for every Regex 1 result, if result is in Regex 2, we ignore it

Downsides: Very inefficient, fails if users use \` as a regular quote, relies on parsing \` specifically (and not \<pre\> \<code\> etc.)

**Comment by peterrus** [comment]
Yeah this one seems quite hard. Maybe someone comes along with a good idea!
